src/Jitter_CodeGen_x86_Fpu_Avx.cpp

```diff
@@ -8,23 +8,61 @@ void CCodeGen_x86::Emit_Fpu_Avx_MemMem(const STATEMENT& statement)
 	auto dst = statement.dst->GetSymbol().get();
 	auto src1 = statement.src1->GetSymbol().get();
 
-	((m_assembler).*(FPUOP::OpEdAvx()))(CX86Assembler::xMM0, CX86Assembler::xMM0, MakeMemoryFpSingleSymbolAddress(src1));
-	m_assembler.VmovssEd(MakeMemoryFpSingleSymbolAddress(dst), CX86Assembler::xMM0);
+	auto dstRegister = PrepareSymbolRegisterDefFpu(dst, CX86Assembler::xMM0);
+	auto src1Register = PrepareSymbolRegisterUseFpuAvx(src1, CX86Assembler::xMM1);
+
+	((m_assembler).*(FPUOP::OpEdAvx()))(dstRegister, dstRegister, CX86Assembler::MakeXmmRegisterAddress(src1Register));
+
+	CommitSymbolRegisterFpuAvx(dst, dstRegister);
+
+}
+
+void CCodeGen_x86::Emit_Fp_Avx_Neg_VarVar(const STATEMENT& statement)
+{
+	auto dst = statement.dst->GetSymbol().get();
+	auto src1 = statement.src1->GetSymbol().get();
+
+	auto tmpIntRegister = CX86Assembler::rAX;
+	auto tmpXMMRegister = CX86Assembler::xMM2;
+	auto tmpXMM2Register = CX86Assembler::xMM3;
+	auto dstRegister = PrepareSymbolRegisterDefFpu(dst, CX86Assembler::xMM0);
+	auto src1Register = PrepareSymbolRegisterUseFpuAvx(src1, CX86Assembler::xMM1);
+
+	m_assembler.MovId(tmpIntRegister, 0x80000000);
+	m_assembler.VmovdVo(tmpXMMRegister, CX86Assembler::MakeRegisterAddress(tmpIntRegister));
+	m_assembler.VxorpsVo(dstRegister, src1Register, CX86Assembler::MakeXmmRegisterAddress(tmpXMMRegister));
+	CommitSymbolRegisterFpuAvx(dst, dstRegister);
+}
+
+void CCodeGen_x86::Emit_Fp_Avx_Abs_VarVar(const STATEMENT& statement)
+{
+	auto dst = statement.dst->GetSymbol().get();
+	auto src1 = statement.src1->GetSymbol().get();
+
+	auto dstRegister = PrepareSymbolRegisterDefFpu(dst, CX86Assembler::xMM0);
+	auto src1Register = PrepareSymbolRegisterUseFpuAvx(src1, CX86Assembler::xMM1);
+	auto tmpXMMRegister = CX86Assembler::xMM2;
+	auto tmpXMM2Register = CX86Assembler::xMM3;
+
+	m_assembler.VxorpsVo(tmpXMMRegister, tmpXMMRegister, CX86Assembler::MakeXmmRegisterAddress(tmpXMMRegister));
```

Comment:
I would avoid using FP operations here to prevent any shenanigans with NaNs and INFs that the PS2 handles differently. Using this sequence should do the job:

```
vpcmpeqd xmm0, xmm0, xmm0
vpsrld xmm0, xmm0, 1   (generate 0x7FFFFFFF)
vandps dstRegister, dstRegister, xmm0
```